src/coreclr/jit/switchrecognition.cpp

```diff
@@ -22,26 +22,33 @@
 //
 PhaseStatus Compiler::optSwitchRecognition()
 {
-// Limit to XARCH, ARM is already doing a great job with such comparisons using
-// a series of ccmp instruction (see ifConvert phase).
-#ifdef TARGET_XARCH
     bool modified = false;
     for (BasicBlock* block = fgFirstBB; block != nullptr; block = block->Next())
     {
-        // block->KindIs(BBJ_COND) check is for better throughput.
+#ifdef TARGET_XARCH
+        // Limit to XARCH, ARM is already doing a great job with such comparisons using
+        // a series of ccmp instruction (see ifConvert phase).
         if (block->KindIs(BBJ_COND) && !block->isRunRarely() && optSwitchDetectAndConvert(block))
         {
             JITDUMP("Converted block " FMT_BB " to switch\n", block->bbNum)
             modified = true;
         }
+#endif
+
+        // See if we can merge BBJ_COND into an existing switch block
```

Comment:
I noticed for tier1-instr code this causes a lot of code expansion -- maybe skip doing this if we're instrumenting?

Reply:
Good point

Reply:
Addressed